tokio/src/runtime/context.rs

```diff
@@ -93,31 +99,54 @@ impl ThreadContext {
         self
     }
 
-    #[cfg(all(feature = "io-driver", not(loom)))]
+    #[cfg(any(
+        feature = "blocking",
+        feature = "dns",
+        feature = "fs",
+        feature = "io-std",
+        feature = "rt-threaded",
+    ))]
+    pub(crate) fn with_blocking_spawner(
+        mut self,
+        blocking_spawner: crate::runtime::blocking::Spawner,
+    ) -> Self {
+        self.blocking_spawner.replace(blocking_spawner);
+        self
+    }
+
     pub(crate) fn io_handle() -> crate::runtime::io::Handle {
+        #[cfg(any(not(feature = "io-driver"), loom))]
```

Comment:
Could you elaborate on why this change was made?

Reply:
Oh. I think it was failing CI with a unused warning? Though I can try reverting it.

Reply:
Should the `fn io_handle` be scoped w/ a `cfg` then? The fns below have had their `cfg` bits removed. I'd be interested in seeing the error.

Reply:
So the way that it's written right now, the field is not optional, it gets swapped out for `()` when these features aren't enabled (apparently). Here's the error with a default cargo check with those changed reverted::

```shell
error[E0308]: match arms have incompatible types
   --> tokio/src/runtime/context.rs:105:21
    |
103 |           CONTEXT.with(|ctx| match *ctx.borrow() {
    |  ____________________________-
104 | |             Some(ref ctx) => ctx.io_handle.clone(),
    | |                              --------------------- this is found to be of type `()`
105 | |             None => None,
    | |                     ^^^^ expected (), found enum `std::option::Option`
106 | |         })
    | |_________- `match` arms have incompatible types
    |
    = note: expected type `()`
               found type `std::option::Option<_>`
```

Reply:
Ah! What if you sub `None` for `Default::default()`?

That said, it is curious that `io_handle()` is required even when the `io-driver` feature isn't enabled.

Reply:
This stems from the way the Handle::Current works, which asks for all the components of ThreadContext, where Handle has all of these, so cfging off the io_handle function will spider out for a lot of other changes. Maybe Default::default() will work...

Reply:
Ok, that works.